src/logic/claims/data.ts

```diff
@@ -191,15 +191,26 @@ async function loadClaims(
   };
 }
 
-function calculateDeadline(claim: Claim) {
+export function calculateDeadline(claim: Pick<Claim, 'status' | 'statusUpdatedAt' | 'dispute'>) {
   const { dispute } = claim;
 
   switch (claim.status) {
     case 'ClaimCreated':
     case 'SettlementProposed':
       return addDays(claim.statusUpdatedAt, 3);
     case 'DisputeCreated':
-      return dispute?.period === 'Appeal' ? dispute.periodEndDate! : null;
+      switch (dispute?.period) {
+        case 'Evidence':
+          // Kleros gives their ruling after the voting period, so we get the voting period end date by adding the vote
+          // period to the evidence period (the commit period is skipped because the sub court does not have hidden votes).
+          return addSeconds(dispute.periodChangedAt, dispute.timesPerPeriod[0]! + dispute.timesPerPeriod[2]!);
```

Comment:
I find these indexes a bit error-prone. Can we maybe have a fields such as `evidence`, 'commit', etc... instead of `timesPerPeriod` ? Alternatively, we could use named constants for indexing the times per period, but I like that a bit less.

Reply:
Yeah I was also on the fence about this. I reckon it will help to change:
```ts
      timesPerPeriod: subCourt.timesPerPeriod.map((time) => time.toNumber()),
```
to:
```ts
      periodTimes: {
        evidence: subCourt.timesPerPeriod[0].toNumber(),
        vote: subCourt.timesPerPeriod[2].toNumber(),
        appeal: subCourt.timesPerPeriod[3].toNumber(),
      },
```
when loading the data so that subsequent uses are at least a bit more obvious.

Reply:
Yeah and if you want to stress out the order of the periods, you can write a comment on the interface type (but I don't think it's necessary).